app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt

```diff
@@ -189,7 +197,7 @@ class AddonsManagementFragment : Fragment(R.layout.fragment_add_ons_management)
                 ),
                 onConfirmButtonClicked = { _, allowInPrivateBrowsing ->
                     if (allowInPrivateBrowsing) {
-                        requireContext().components.addonManager.setAddonAllowedInPrivateBrowsing(
+                        weakApplicationContext.get()?.components?.addonManager?.setAddonAllowedInPrivateBrowsing(
```

Comment:
I would prefer if we could use [runIfFragmentIsAttached](https://github.com/mozilla-mobile/fenix/blob/ea2bb0df75c407e98618838cf39f90ee8bc421d9/app/src/main/java/org/mozilla/fenix/addons/Extensions.kt#L32) extension function as using `requireActivity()` when the fragment is initialized could cause crashes :)